File: core/src/mindustry/client/ui/FeaturesDialog.java
package mindustry.client.ui;

import arc.Core;
import arc.scene.ui.ScrollPane;
import mindustry.ui.dialogs.BaseDialog;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

public class FeaturesDialog extends BaseDialog {

    public FeaturesDialog() {
        super("Features and Documentation");
        cont.pane(StupidMarkupParser.format(Core.files.internal("features").readString("UTF-8"))).growX().get().setScrollingDisabled(true, false);
        addCloseButton();
    }
}

File: core/src/mindustry/client/ui/FeaturesDialog.kt
package mindustry.client.ui

import arc.*
import arc.scene.ui.*
import mindustry.client.*
import mindustry.input.*
import mindustry.ui.dialogs.*

object FeaturesDialog : BaseDialog("@client.features") {
    override fun show(): Dialog {
        cont.clear()
        buttons.clear()
        clearListeners()

        var str = Core.files.internal("features").readString("UTF-8")
        str = str.replace("\\{\\w+}".toRegex()) { res ->
            val value = res.value.removeSurrounding("{", "}")
            if (value == "p") return@replace ClientVars.clientCommandHandler.prefix // {p} becomes the client command prefix
            try {
                val bind = Binding.valueOf(value)
                Core.keybinds[bind].key.value
            } catch (ignored: Exception) { // If this isn't a keybinding, we will keep it as is.
                return@replace res.value
            }
        }
        cont.pane(StupidMarkupParser.format(str)).growX().get()
            .setScrollingDisabled(true, false)
        addCloseButton()

        return super.show()
    }
}